packages/pwa-buildpack/lib/Utilities/serve.js

```diff
@@ -33,6 +35,24 @@ module.exports = async function serve(dirname) {
         }
     );
 
+    if (customHttpsSettings.key && customHttpsSettings.cert) {
+        const { key, cert } = customHttpsSettings;
+        if (fs.existsSync(key) && fs.existsSync(cert)) {
+            prettyLogger.info(
+                'Custom key and cert paths provided, creating HTTPS server.'
+            );
+            const ssl = {
+                key: fs.readFileSync(key, 'utf8'),
+                cert: fs.readFileSync(cert, 'utf8')
+            };
+            upwardServerOptions.https = ssl;
```

Comment:
If I read this file correctly it looks like it is possible for this setting to get overwritten by the [conditional block](
https://github.com/magento/pwa-studio/blob/c64cb3535b5b7e7adabec608f6c372f4e5d3cac6/packages/pwa-buildpack/lib/Utilities/serve.js#L78-L102) further on in the file.

I might add a condition around this line (currently 93):

```es6
if (!upwardServerOptions.https) {
  upwardServerOptions.https = ssl;
}
```

This way, if someone has specified their own key/cert, we won't overwrite it when configuring the `customOrigin` settings.

Reply:
Thanks for feedback, added it ;)